src/observable/ConnectableObservable.ts

```diff
@@ -17,28 +17,35 @@ export class ConnectableObservable<T> extends Observable<T> {
     return this._getSubject().subscribe(subscriber);
   }
 
-  _getSubject() {
-    const subject = this.subject;
+  protected _getSubject() {
+    const subject = this._subject;
     if (subject && !subject.isUnsubscribed) {
       return subject;
     }
-    return (this.subject = this.subjectFactory());
+    return (this._subject = this.subjectFactory());
   }
 
   connect(): Subscription {
     const source = this.source;
-    let subscription = this.subscription;
+    let subscription = this._subscription;
     if (subscription && !subscription.isUnsubscribed) {
       return subscription;
     }
     subscription = source.subscribe(this._getSubject());
     subscription.add(new ConnectableSubscription(this));
-    return (this.subscription = subscription);
+    return (this._subscription = subscription);
   }
 
   refCount(): Observable<T> {
     return new RefCountObservable(this);
   }
+
+  // This method is opened for `ConnectableSubscription`.
+  // Not to call from others.
+  _closeSubscription(): void {
```

Comment:
Once it's published as public, regardless of name starts with _ or not it is marked as consumer-usable. In my opinion, for this case I'd revert back to original code to expose `subject` / `subscription` directly for simple changes. Or either could think of more strict dependencies but initial assumption it'll require amount of change and might be cumbersome impact performance.